replica/database.cc

```diff
@@ -1723,10 +1723,31 @@ future<mutation> database::apply_counter_update(schema_ptr s, const frozen_mutat
   }));
 }
 
+// #9919 etc. The initiative to wrap exceptions here
+// causes a bunch of problems with (implicit) call sites
+// catching timed_out_error (not checking is_timeout_exception).
+// Fixing the call sites is a good idea, but it is also hard
+// to verify. This workaround should ensure we take the
+// correct code paths in all cases, until we can clean things up
+// proper.
+class wrapped_timed_out_error : public timed_out_error {
```

Comment:
maybe derive from both timed_out_error and std::runtime_error?

Reply:
Both derive (non-virtual) from std::exception -> not a good idea.

Reply:
okay